math/binarymath/checkisnumberpoweroftwo.go

```diff
@@ -0,0 +1,37 @@
+// checkisnumberpoweroftwo.go
+// description: Is the number a power of two
+// details:
+// Checks if a number is a power of two- [Power of two](https://en.wikipedia.org/wiki/Power_of_two)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see checkisnumberpoweroftwo_test.go
+
+package binarymath
+
+import (
+	"math"
+)
+
+func IsPowerOfTwoBinaryMethod(x int) bool {
+	return x > 0 && (x&(x-1)) == 0
+}
+
+func IsPowerOfTwoUseCycleAndLeftShift(number uint) bool {
```

Comment:
```suggestion
func IsPowerOfTwoLeftShift(number uint) bool {
```

Reply:
I forgot to mention this, but add an explanation for this too.